cmd/lotus/daemon.go

```diff
@@ -463,6 +465,16 @@ func ImportChain(ctx context.Context, r repo.Repo, fname string, snapshot bool)
 		l = st.Size()
 	}
 
+	if path.Ext(fname) == ".zst" {
+		zr := zstd.NewReader(rd)
+		defer func() {
+			if err := zr.Close(); err != nil {
+				log.Errorw("closing zstd reader", "error", err)
+			}
+		}()
+		rd = zr
+	}
+
```

Comment:
This can be done smarter / more statelessly:

- wrap a `bufio.Reader`
- `r.Peek()` the first  4 bytes
- match  [`"\xB5\x2F\xFD"` at position 1 ( not 0 )](https://github.com/file/file/blob/FILE5_38/magic/Magdir/compress#L308-L323)
- profit accordingly